xbmc/GUIPassword.h

```diff
@@ -64,20 +65,24 @@ class CGUIPassword : public ISettingCallback
   void LockSources(bool lock);
   void RemoveSourceLocks();
   bool IsDatabasePathUnlocked(const std::string& strPath, VECSOURCES& vecSources);
-  /*! \brief Tests if the user is allowed to access the path by looking up the matching Mediasource
-   \param strPath The folder path to access
-   \param strType The type of share being accessed, e.g. "music", "video", etc. See CSettings::UpdateSources()
-   \return If access is granted, returns \e true
-   */
-  bool IsMediaPathUnlocked(const std::string& strPath, const std::string& strType);
+
+  std::string getMediasourcePath() { return strMediasourcePath; }
```

Comment:
Should this return a copy or a reference?

Reply:
good question ;) i don't know. but i'd say copy if someday the value needs changing in that getter

Reply:
`const std::string&`, please. If somebody needs to modify the return value, they can create their own copy.

Reply:
And the method itself should be `const` and method names should start with a capital letter.
=> `const std::string& GetMediasourcePath() const { return strMediasourcePath; }`